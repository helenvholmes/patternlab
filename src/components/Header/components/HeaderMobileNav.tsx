import {
  Box,
  chakra,
  Flex,
  Spacer,
  useMultiStyleConfig,
} from "@chakra-ui/react";

import { siteNavLinks, upperNavLinks } from "../utils/headerUtils";
import Link from "../../Link/Link";
import Logo from "../../Logo/Logo";
import List from "../../List/List";
import SimpleGrid from "../../Grid/SimpleGrid";
import { Icon } from "../../Icons/Icon";

/**
 * This component renders the navigational list of links used to navigate
 * NYPL.org for mobile devices.
 */
const HeaderMobileNav = chakra(() => {
  const styles = useMultiStyleConfig("HeaderMobileNav", {});
  const listItems = siteNavLinks.map(({ href, text }) => (
    <Link href={href} key={text}>
      {text}
    </Link>
  ));

  return (
    <Box __css={styles}>
      <Flex>
        <Box>
          <Logo
            aria-label="NYPL Header Logo"
            decorative={false}
            name="nyplTextWhite"
            size="xsmall"
            title="NYPL Header Logo"
            __css={styles.logo}
          />
        </Box>
        <Spacer />
        <nav aria-label="Header mobile links">
          <List
            id="header-mobile-nav"
            listItems={listItems}
            noStyling
            type="ul"
            __css={{
              ...styles.sideNav,
              li: { marginBottom: "unset !important" },
            }}
          />
        </nav>
      </Flex>
      <SimpleGrid data-testid="bottomLinks" __css={styles.bottomLinks}>
        <Link
          href={upperNavLinks.libraryCard.href}
          borderTop="1px solid rgb(54, 54, 54)"
          borderRight="1px solid rgb(54, 54, 54)"
          gridColumn="1 / span 1"
        >
          <Icon
            align="left"
            color="ui.white"
            name="decorativeLibraryCard"
            size="large"
          />
          {upperNavLinks.libraryCard.text}
        </Link>
        <Link
          href={upperNavLinks.emailUpdates.href}
          borderTop="1px solid rgb(54, 54, 54)"
          gridColumn="2 / span 1"
        >
          <Icon
            align="left"
            color="ui.white"
            name="decorativeEnvelope"
            size="large"
          />
          {upperNavLinks.emailUpdates.text}
        </Link>
        <Link
          href={upperNavLinks.shop.href}
          borderTop="1px solid rgb(54, 54, 54)"
          gridColumn="1 / span 2"
        >
          <Icon
            align="left"
            color="ui.white"
            name="decorativeShoppingBag"
            size="large"
          />
          {upperNavLinks.shop.text} NYPL
        </Link>
        <Link href={upperNavLinks.donate.href} gridColumn="1 / span 2">
          {upperNavLinks.donate.text.toUpperCase()}
        </Link>
      </SimpleGrid>
    </Box>
  );
});

export default HeaderMobileNav;
