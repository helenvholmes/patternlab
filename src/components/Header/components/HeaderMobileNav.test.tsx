import { render, screen, within } from "@testing-library/react";

import { axe } from "jest-axe";
import * as renderer from "react-test-renderer";

import HeaderMobileNav from "./HeaderMobileNav";


describe("HeaderMobileNav Accessibility", () => {
  it("passes axe accessibility test", async () => {
    const { container } = render(<HeaderMobileNav />);
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("HeaderMobileNav", () => {
  beforeEach(() => {
    render(<HeaderMobileNav />);
  });

  it("renders the NYPL logo", () => {
    const logo = screen.getByTitle("NYPL Header Logo");
    expect(logo).toBeInTheDocument();
  });

  it("renders a nav list", () => {
    const navList = screen.getByRole("navigation");
    const links = within(navList).getAllByRole("link");

    expect(links).toHaveLength(7);
    expect(links[0]).toHaveTextContent("Books/Music/Movies");
    expect(links[1]).toHaveTextContent("Research");
    expect(links[2]).toHaveTextContent("Education");
    expect(links[3]).toHaveTextContent("Events");
    expect(links[4]).toHaveTextContent("Connect");
    expect(links[5]).toHaveTextContent("Give");
    expect(links[6]).toHaveTextContent("Get Help");

    links.forEach((link) => link.click());
  });

  it("renders the bottom links", () => {
    const bottomGrid = screen.getByTestId("bottomLinks");
    const links = within(bottomGrid).getAllByRole("link");

    expect(links).toHaveLength(4);
    expect(links[0]).toHaveTextContent("Get A Library Card");
    expect(links[1]).toHaveTextContent("Get Email Updates");
    expect(links[2]).toHaveTextContent("Shop NYPL");
    expect(links[3]).toHaveTextContent("DONATE");

    links.forEach((link) => link.click());
  });

  it("renders the UI snapshot correctly", () => {
    const headerMobileNav = renderer.create(<HeaderMobileNav />).toJSON();

    expect(headerMobileNav).toMatchSnapshot();
  });
});
