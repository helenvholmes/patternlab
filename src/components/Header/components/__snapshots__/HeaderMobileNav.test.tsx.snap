// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMobileNav renders the UI snapshot correctly 1`] = `
<div
  className="css-0"
>
  <div
    className="css-k008qs"
  >
    <div
      className="css-0"
    >
      <svg
        aria-hidden={false}
        aria-label="NYPL Header Logo"
        className="chakra-icon css-1vgifmd"
        data-file-name="SvgLogoNyplTextWhite"
        focusable={false}
        role="img"
        title="NYPL Header Logo"
      />
    </div>
    <div
      className="css-17xejub"
    />
    <nav
      aria-label="Header mobile links"
    >
      <ul
        className="css-oxl4r2"
        id="header-mobile-nav"
      >
        <li>
          <a
            className="chakra-link css-1xdhyk6"
            href="https://www.nypl.org/books-music-movies"
            rel={null}
            target={null}
          >
            Books/Music/Movies
          </a>
        </li>
        <li>
          <a
            className="chakra-link css-1xdhyk6"
            href="https://www.nypl.org/research"
            rel={null}
            target={null}
          >
            Research
          </a>
        </li>
        <li>
          <a
            className="chakra-link css-1xdhyk6"
            href="https://www.nypl.org/education"
            rel={null}
            target={null}
          >
            Education
          </a>
        </li>
        <li>
          <a
            className="chakra-link css-1xdhyk6"
            href="https://www.nypl.org/events"
            rel={null}
            target={null}
          >
            Events
          </a>
        </li>
        <li>
          <a
            className="chakra-link css-1xdhyk6"
            href="https://www.nypl.org/connect"
            rel={null}
            target={null}
          >
            Connect
          </a>
        </li>
        <li>
          <a
            className="chakra-link css-1xdhyk6"
            href="https://www.nypl.org/give"
            rel={null}
            target={null}
          >
            Give
          </a>
        </li>
        <li>
          <a
            className="chakra-link css-1xdhyk6"
            href="https://www.nypl.org/get-help"
            rel={null}
            target={null}
          >
            Get Help
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="css-1xdhyk6"
    data-testid="bottomLinks"
  >
    <a
      className="chakra-link css-a4nb38"
      href="https://www.nypl.org/library-card/new"
      rel={null}
      target={null}
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1k5hm9w"
        data-file-name="SvgDecorativeLibraryCard"
        focusable={false}
        role="img"
        title="decorativeLibraryCard icon"
      />
      Get A Library Card
    </a>
    <a
      className="chakra-link css-ucg3r3"
      href="https://pub.email.nypl.org/subscriptioncenter"
      rel={null}
      target={null}
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1k5hm9w"
        data-file-name="SvgDecorativeEnvelope"
        focusable={false}
        role="img"
        title="decorativeEnvelope icon"
      />
      Get Email Updates
    </a>
    <a
      className="chakra-link css-xnpdth"
      href="https://shop.nypl.org/?utm_campaign=NYPLHeaderButton&utm_source=nypl.org&utm_medium=referral"
      rel={null}
      target={null}
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1k5hm9w"
        data-file-name="SvgDecorativeShoppingBag"
        focusable={false}
        role="img"
        title="decorativeShoppingBag icon"
      />
      Shop
       NYPL
    </a>
    <a
      className="chakra-link css-tlhajk"
      href="https://www.nypl.org/donate-button"
      rel={null}
      target={null}
    >
      DONATE
    </a>
  </div>
</div>
`;
